### PR TITLE
add _sqlite3 module

### PR DESCRIPTION
**Pull request by tungol**
This aligns with the implementation while giving greater fidelity to runtime naming and inheritance

Related to https://github.com/python/typeshed/issues/3968 and https://github.com/python/typeshed/issues/11141

### CODE REVIEW

File: stdlib/VERSIONS [hunk 1]
```diff
@@ -46,6 +46,7 @@ _pydecimal: 3.5-
 _random: 2.7-
 _sitebuiltins: 3.4-
 _socket: 3.0-  # present in 2.7 at runtime, but not in typeshed
+_sqlite3: 2.7-
```
Comment (JelleZijlstra) [inline stdlib/VERSIONS:49]:
```suggestion
_sqlite3: 3.0-
```
We just merged a PR removing all the 2.7 lines.

Also, general note: I was away for a few weeks and I see you have a number of open PRs that haven't been merged yet. I'm just glancing through my notifications at the moment, but when I find some time I'll try to go over open nondraft typeshed PRs to review what I can.

Comment (tungol) [reply stdlib/VERSIONS:49]: Appreciated, but no rush

**Comment by JelleZijlstra** [comment]
This PR also has a lot of merge conflicts. I can review it if the conflicts are resolved and it's taken out of draft.

**Comment by tungol** [comment]
Thanks, @JelleZijlstra ! This one should be tidied up and good to go now

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**